### PR TITLE
Tech Debt

### LINKED ISSUE #15: system tray Icon still appearing after application close
If simbridge is not closed via the systray icon then the systray process is not properly killed. References to properly kill the process https://github.com/felixhao28/node-systray/issues/14

### PR DESCRIPTION
**Pull request by Lucky38i**
Resolves #15 

- adjust eslintignore
- remove prettier, adjust linting scripts
- Added experimental shutdown listener from https://github.com/flybywiresim/simbridge/tree/fix/shutdown
- Moved msfs listener to it's own service
- Ip utility is now a service
- systray now properly constructed as a service

### CODE REVIEW

File: apps/server/src/utilities/shutdown.service.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-  constructor(private systrayService: SysTrayService) {}
+    private shutdownListener$: Subject<void> = new Subject();
 
-  onApplicationShutdown(signal: string) {
-      this.logger.log(`Application shutting down with signal: ${signal}`);
-      // Handle all graceful kill events
-      this.systrayService.kill();
-  }
+    onModuleDestroy() {
+        this.logger.log('ShutDown Service being destroyed');
```
Comment (Nufflee) [inline apps/server/src/utilities/shutdown.service.ts:11]:
```suggestion
        this.logger.log('Destroying ShutDown Service');
```

Just to match the SysTray service.

Comment (Lucky38i) [reply apps/server/src/utilities/shutdown.service.ts]: resolved [88f4d09](https://github.com/flybywiresim/simbridge/pull/58/commits/88f4d09fec86b16a99cdf417d2931e91fb89e74d)

File: apps/server/src/utilities/shutdown.service.ts [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+import { Subject } from 'rxjs';
 
 @Injectable()
-export class ShutDownService implements OnApplicationShutdown {
-  private readonly logger = new Logger(ShutDownService.name);
+export class ShutDownService implements OnModuleDestroy {
+    private readonly logger = new Logger(ShutDownService.name);
 
-  constructor(private systrayService: SysTrayService) {}
+    private shutdownListener$: Subject<void> = new Subject();
```
[the remark this answers is not in the document]
Comment (Lucky38i) [reply apps/server/src/utilities/shutdown.service.ts]: resolved [88f4d09](https://github.com/flybywiresim/simbridge/pull/58/commits/88f4d09fec86b16a99cdf417d2931e91fb89e74d)

File: apps/server/src/utilities/systray.service.ts [hunk 3]
```diff
@@ -1,96 +1,106 @@
...
-      this.sysTray.kill();
-  }
+    private manageConsole() {
+        if (this.hidden) showConsole();
+        else hideConsole();
+        this.hidden = !this.hidden;
+    }
+
+    onApplicationShutdown(signal?: string) {
+        this.logger.log(`Shutdown signal: ${signal} received, Shutting down Systray Service`);
```
[the remark this answers is not in the document]
Comment (Lucky38i) [reply apps/server/src/utilities/systray.service.ts]: resolved [88f4d09](https://github.com/flybywiresim/simbridge/pull/58/commits/88f4d09fec86b16a99cdf417d2931e91fb89e74d)

**Review by Nufflee — APPROVED** [review]

**Pull request merged**